src/main/java/core/basesyntax/ArrayList.java

```
@@ -1,48 +1,121 @@
+            double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+            this.capacity = (int) newCapacity;
+            T[] newElementsData = (T[]) new Object[capacity];
+            System.arraycopy(elementsData, 0, newElementsData, 0, size);
+            elementsData = newElementsData;
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size() == capacity) {
+            double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+            this.capacity = (int) newCapacity;
+            T[] newElementsData = (T[]) new Object[capacity];
+            System.arraycopy(elementsData, 0, newElementsData, 0, size);
+            elementsData = newElementsData;
+        }
+        if (index >= 0 && index <= size) {
+            for (int i = elementsData.length; i > 0; i--) {
+                elementsData[i] = elementsData[i - 1];
+                if (i == index) {
+                    elementsData[i] = value;
+                    size++;
+                    break;
+                }
+            }
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            for (int i = 0; i < elementsData.length; i++) {
+                if (i == index) {
+                    return elementsData[i];
+                }
+            }
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            elementsData[index] = value;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
```

You should throw the exception inside an else block or after an if block that checks the index. Otherwise, the exception will be thrown every time the method is called, regardless of whether the index is valid or not.